packages/extension/src/companion/CompanionContextMenu.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { ReactElement, useState } from 'react';
+import CommentIcon from '@dailydotdev/shared/icons/comment.svg';
+import ShareIcon from '@dailydotdev/shared/icons/share.svg';
+import FlagIcon from '@dailydotdev/shared/icons/flag.svg';
+import FeedbackIcon from '@dailydotdev/shared/icons/feedback.svg';
+import EyeIcon from '@dailydotdev/shared/icons/eye.svg';
+import { Item, Menu } from '@dailydotdev/react-contexify';
+import RepostPostModal from '@dailydotdev/shared/src/components/modals/ReportPostModal';
+import { BootData } from './common';
+import DisableCompanionModal from './DisableCompanionModal';
+
+export default function CompanionContextMenu({
+  postData,
+  onMessage,
+  onReport,
+  onBlockSource,
+  onDisableCompanion,
+}: {
+  postData: BootData;
+  onMessage?: (message: string, timeout?: number) => Promise<unknown>;
+  onReport;
+  onBlockSource;
+  onDisableCompanion;
+}): ReactElement {
```

Comment:
I think we can create an interface for this so we can also explicitly define the parameters.